src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2506,6 +2509,11 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 srcCount += BuildDelayFreeUses(emitOp2, emitOp1);
                 srcCount += emitOp3->isContained() ? BuildOperandUses(emitOp3) : BuildDelayFreeUses(emitOp3, emitOp1);
 
+                if (intrinsicTree->OperIsEmbRoundingEnabled() && !intrinsicTree->Op(4)->IsCnsIntOrI())
+                {
+                    srcCount += BuildDelayFreeUses(intrinsicTree->Op(4), emitOp1);
```

Comment:
I don't think we need `BuildDelayFreeUses` here. `op(4)` is an integer and so will go in a GPR. `op1` is a vector and so will go in a SIMD register

Reply:
I have limited understand within LSRA domain, it would be much appreciated if you could point me to the reasonable function to use...

Reply:
For this case you should just need `BuildOperandUses(intrinsicTree->Op(4))`

The general premise is:
* `BuildOperandUses` -- safe to use when there aren't any restrictions
* `BuildDelayFreeUses` -- safe to use when you need to restrict the register set, typically used for RMW nodes

In this case, `FMA` is an `RMW` node. However, the register sets used by `op4` (simd) and `op1` (general-purpose) don't overlap, so we don't need to concern ourselves with `op4` being delay free.

Reply:
Thanks for the explanation!